src/main/java/com/redhat/cloud/notifications/routers/UserConfigService.java

```diff
@@ -90,16 +90,26 @@
 
         final List<Multi<Boolean>> subscriptionRequests = new ArrayList<>();
 
-        values.bundles.forEach((bundle, bundleSettingsValue) ->
-                bundleSettingsValue.applications.forEach((application, applicationSettingsValue) ->
+        values.bundles.forEach((bundleName, bundleSettingsValue) ->
+                bundleSettingsValue.applications.forEach((applicationName, applicationSettingsValue) ->
                 applicationSettingsValue.notifications.forEach((emailSubscriptionType, value) -> {
                     if (value) {
-                        subscriptionRequests.add(emailSubscriptionResources.subscribe(
-                                account, name, bundle, application, emailSubscriptionType
-                        ).toMulti());
+                        subscriptionRequests.add(
+                                applicationResources.getApplication(bundleName, applicationName)
```

Comment:
Checking the app existence with a `SELECT` before the `INSERT` will work, but we could also do that check directly into the `INSERT` query which would be a bit better in terms of performances.

Reply:
This is just a workaround, when we migrate to using the id's rather than the names we will be able to remove the check.